### ISSUE TITLE
np.random.uniform type_code_ == kNDArrayHandle (7 vs. 11)

### ISSUE THREAD

**Issue by leezu** [apache/incubator-mxnet#18254]
`numpy.random.uniform` accepts 
```
high : float or array_like of floats
        Upper boundary of the output interval.  All values generated will be
        less than high.  The default value is 1.0.
```

but mxnet crashes when passed an array_like.

```
>>> import mxnet as mx
>>> mx.npx.set_np()
>>> mx.np.random.uniform((64,512))
terminate called after throwing an instance of 'dmlc::Error'
  what():  [02:56:06] include/mxnet/runtime/packed_func.h:521: Check failed: type_code_ == kNDArrayHandle (7 vs. 11) :  expected
```

```
>>> import mxnet as mx
>>> mx.np.random.uniform((64,512))
terminate called after throwing an instance of 'dmlc::Error'
  what():  [02:56:06] include/mxnet/runtime/packed_func.h:521: Check failed: type_code_ == kNDArrayHandle (7 vs. 11) :  expected
```

```
>>> import numpy as np
>>> np.random.uniform((64,512))
array([  5.60511983, 280.31067919])
```

If array_like is unsupported in mxnet, at least there shouldn't be a crash but rather an error message explaining the lack of support.

**Comment by Tommliu** [comment]
In deepnumpy, we don't support the array_like input for np.random.uniform. It is different from the official numpy. You can refer from the docs (mx.np.random.uniform)
`----------
    low : float, ndarray, optional
        Lower boundary of the output interval.  All values generated will be
        greater than or equal to low.  The default value is 0.
    high : float, ndarray, optional
        Upper boundary of the output interval.  All values generated will be
        less than high.  The default value is 1.0.`

**Comment by leezu** [comment]
Yes, we don't support it currently. But it shouldn't cause "termination".

When adding the error message, is it feasible to ensure that it covers all operators instead of only the uniform sampling operator? I suspect there may be more operators where the numpy support is not exact and where a similar crash may occur?
